docs/features/search/declarative-integration.md

```diff
@@ -0,0 +1,229 @@
+
+/** @alpha */
+export const TechDocsSearchResultListItemExtension =
+  createSearchResultListItemExtension({
+    id: 'techdocs',
+    configSchema: createSchemaFromZod(z =>
+      z.object({
+        noTrack: z.boolean().default(false),
+        lineClamp: z.number().default(5),
+      }),
+    ),
+    predicate: result => result.type === 'techdocs',
+    component: async ({ config }) => {
+      const { TechDocsSearchResultListItem } = await import(
+        './components/TechDocsSearchResultListItem'
+      );
+      return props => <TechDocsSearchResultListItem {...props} {...config} />;
+    },
+  });
+```
+
+In snippet above, a plugin developer is providing a custom component for rendering search results of type "techdocs". The custom result item extension will be enabled by default once the `@backstage/plugin-techdocs` package is installed, that means adopters don't have to enable the extension manually via configuration file.
+
+When a Backstage adopter doesn't want to use the custom `TechDocs` search result item after installing the `TechDocs` plugin, they could disable it via Backstage configuration file:
+
+```yaml
+# app-config.yaml
+app:
+  experimental:
+    packages: 'all'
+  extensions:
+    - plugin.search.result.item.techdocs: false # ✨
+```
+
+Because a configuration schema was provided to the extension factory, Backstage adopters will be able to customize `TechDocs` search results **line clamp** that defaults to 3 and also **disable automatic analytic events tracking**:
+
+```yaml
+# app-config.yaml
+app:
+  experimental:
+    packages: 'all' # enable packages auto discovery
+  extensions:
+    - plugin.search.result.item.techdocs:
+        config: # ✨
+          noTrack: true
+          lineClamp: 3
+```
+
+The `createSearchResultItemExtension` function return an Backstage's extension representation as follows:
+
+```ts
+{
+  "$$type": "@backstage/Extension", // [1]
+  "id": "plugin.search.result.item.techdocs", // [2]
+  "at": "plugin.search.page/items", // [3]
+  "inputs": {} // [4️]
+  "output": { // [5️]
+    "item": {
+      "$$type": "@backstage/ExtensionDataRef",
+      "id": "plugin.search.result.item.data",
+      "config": {}
+    }
+  },
+  "configSchema": { // [6️]
+    "schema": {
+      "type": "object",
+      "properties": {
+        "noTrack": {
+          "type": "boolean",
+          "default": false
+        },
+        "lineClamp": {
+          "type": "number",
+          "default": 5
+        }
+      },
+      "additionalProperties": false,
+      "$schema": "http://json-schema.org/draft-07/schema#"
+    }
+  },
+  "disabled": false, // [7️]
+}
+```
+
+In this object, you can see exactly what will happen once the custom extension is installed:
+
+- **[1] $$type**: declares that the object represents an extension;
+- **[2] id**: Is an unique identification for the extension, the `plugin.search.result.item.techdocs` is the key used to configure the extension in the `app-config.yaml` file;
```

Comment:
Is a unique

Reply:
Fixed 🙏🏻 : https://github.com/backstage/backstage/pull/20328/commits/8558c919cb0ee555b420cbc09d2a1de33be7917d